e2e/tests/transfer/authz_test.go

```diff
@@ -248,7 +250,11 @@ func (suite *AuthzTransferTestSuite) TestAuthz_InvalidTransferAuthorizations() {
 			}
 
 			resp := suite.BroadcastMessages(context.TODO(), chainA, granteeWallet, msgExec)
-			suite.AssertTxFailure(resp, ibcerrors.ErrInsufficientFunds)
+			if testvalues.IbcErrorsFeatureReleases.IsSupported(chainAVersion) {
+				suite.AssertTxFailure(resp, ibcerrors.ErrInsufficientFunds)
+			} else {
+				suite.AssertTxFailure(resp, sdkerrors.ErrInsufficientFunds)
+			}
```

Comment:
I'm a bit worried now about the sustainability of this approach, the tests will gradually get messier and messier and harder to work with with all these special cases.

I think there may be some work we can do in the future to come up with an alternate way of handling these changes.

Reply:
opened #3777

Reply:
I agree. Let's open an issue and discuss in one of the engineering calls.